rspace/src/main/scala/coop/rchain/rspace/history/package.scala

```diff
@@ -51,8 +51,9 @@ package object history {
       }
     }
 
-  def lookup[T, K, V](store: ITrieStore[T, K, V], branch: Branch, key: K)(
-      implicit codecK: Codec[K]): Option[V] = {
+  private[this] def lookup[T, K, V](store: ITrieStore[T, K, V],
+                                    root: Either[Branch, Blake2b256Hash],
```

Comment:
@dzajkowski @lukasz-golebiewski: How do you both feel about this use of `Either`?  

While I tend not to mostly use `Either` for error-reporting, this seems like a reasonable approach to avoiding code duplication while exporting an overloaded definition for the two different ways to look things up.

Reply:
alternative would be something along these lines?
```
def lookup[T, K, V](store: ITrieStore[T, K, V], rootHash: Blake2b256Hash, key: K)(
    implicit codecK: Codec[K]): Option[V] = 
store.withTxn(store.createTxnRead()) { (txn: T) => lookup(txn, store, rootHash, key) }

def lookup[T, K, V](store: ITrieStore[T, K, V], branch: Branch, key: K)(
     implicit codecK: Codec[K]): Option[V] = 
store.withTxn(store.createTxnRead()) { (txn: T) => lookup(store, store.getRoot(txn, branch), key) }

```

Reply:
After some thinking I decided to apply changes proposed by @dzajkowski - both implementations will work well, however we also have lookup method that accepts sequence of keys. And I suppose we will save some ticks if all of them will be done inside one transaction.